src/transport/PeerConnectionState.h

```diff
@@ -98,7 +101,20 @@ class PeerConnectionState
         mPeerNodeId         = kUndefinedNodeId;
         mSendMessageIndex   = 0;
         mLastActivityTimeMs = 0;
-        mSecureSession.Reset();
+        mSenderSecureSession.Reset();
+        mReceiverSecureSession.Reset();
+    }
+
+    CHIP_ERROR EncryptBeforeSend(const uint8_t * input, size_t input_length, uint8_t * output, PacketHeader & header,
```

Comment:
Could become const once https://github.com/project-chip/connectedhomeip/issues/4893 is fixed.  So could DecryptOnReceive.

Reply:
The latest commit should fix #4893